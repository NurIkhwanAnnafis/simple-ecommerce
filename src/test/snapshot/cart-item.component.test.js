import React from 'react';
import { act } from "react-dom/test-utils"
import { shallow, mount, render } from 'enzyme';
import CartItem from '../../components/cart-item/cart-item.component';

let dummyProps = null,
    dummyComponent = null,
    result = null;

beforeEach(() => {
  dummyProps = { imageUrl: "https://i.ibb.co/ZYW3VTp/brown-brim.png", name: "Brown Brim", price: 25, quantity: 4 }
})

afterEach(() => {
  dummyProps = null;
  dummyComponent = null;
  result = null;
})

it('should to render cart item component', () => {
  dummyComponent = render(<CartItem item={dummyProps} />);

  act(() => {
    result = dummyComponent;
  })

  expect(result).toMatchSnapshot()
})

it('should to render cart item component with value(props)', () => {
  expect.assertions(9)
  dummyComponent = shallow(<CartItem item={dummyProps} />);

  act(() => {
    result = dummyComponent;
  })

  //parent component
  expect(result.props().className).toEqual('cart-item');

  //child component level 1
  expect(result.props().children[0].type).toEqual('img');
  expect(result.props().children[0].props.src).toEqual('https://i.ibb.co/ZYW3VTp/brown-brim.png');

  expect(result.props().children[1].type).toEqual('div');
  expect(result.props().children[1].props.className).toEqual('item-details');

  //child component level 2
  expect(result.props().children[1].props.children[0].type).toEqual('span');
  expect(result.props().children[1].props.children[0].props.children).toEqual('Brown Brim');
  expect(result.props().children[1].props.children[1].type).toEqual('span');
  expect(result.props().children[1].props.children[1].props.children).toEqual('4 x $25');
})