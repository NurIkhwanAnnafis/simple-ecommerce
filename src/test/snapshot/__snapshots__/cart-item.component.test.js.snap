// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should to render cart item component 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "cart-item",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "alt": "item",
          "src": "https://i.ibb.co/ZYW3VTp/brown-brim.png",
        },
        "children": Array [],
        "name": "img",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "item-details",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "name",
              },
              "children": Array [
                Node {
                  "data": "Brown Brim",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {
                  "class": "price",
                },
                "children": Array [
                  Node {
                    "data": "4 x $25",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Node {
              "attribs": Object {
                "class": "price",
              },
              "children": Array [
                Node {
                  "data": "4 x $25",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {
                  "class": "name",
                },
                "children": Array [
                  Node {
                    "data": "Brown Brim",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "alt": undefined,
          "src": undefined,
        },
        "x-attribsPrefix": Object {
          "alt": undefined,
          "src": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "item-details",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "name",
            },
            "children": Array [
              Node {
                "data": "Brown Brim",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {
                "class": "price",
              },
              "children": Array [
                Node {
                  "data": "4 x $25",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Node {
            "attribs": Object {
              "class": "price",
            },
            "children": Array [
              Node {
                "data": "4 x $25",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "class": "name",
              },
              "children": Array [
                Node {
                  "data": "Brown Brim",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "alt": "item",
            "src": "https://i.ibb.co/ZYW3VTp/brown-brim.png",
          },
          "children": Array [],
          "name": "img",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "alt": undefined,
            "src": undefined,
          },
          "x-attribsPrefix": Object {
            "alt": undefined,
            "src": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
